### DISCUSSION TITLE
API CEP (java) sempre retorna 403

### DISCUSSION THREAD

**Discussion by alexandrelinux** [BrasilAPI/BrasilAPI#446] [General]
Antes de mais nada, parabens pelo projeto, sou desenvolvedor em java, e quando uso a API em java, passo o numero do CEP retorna sempre erro 403, ou seja, é um erro de acesso proibido, nao vi em nenhum lugar que precisaria de fazer algum cadastro para pegar algum ticker de autorização, ou que estou fazendo errado, estou seguindo esse exemplo abaixo.

        BrasilAPI.setEnableLog(true);
        BrasilAPI.setEnableCache(true);
        BrasilAPI.setCacheTimeMinutes(10L);
        CEP cep = BrasilAPI.cep("53140240");

Agradeço pela ajuda.

**Comment by RodriAndreotti** [comment]
Boa tarde, @alexandrelinux!
Beleza?
Não precisa de cadastro, pode ser algo relacionado a User-Agent, tentou enviar algum para ver se funciona?

**Comment by LorhanSohaky** [comment]
@alexandrelinux , conseguiu resolver o problema?

**Comment by thizanchettin** [comment]
Bom dia. 

Sei que é uma postagem antiga, mas eu caí no mesmo problema ontem, então acredito que vale a pena deixar registrado aqui que a solução proposta pelo @RodriAndreotti deu certo, assim se mais alguém vier parar aqui vai ter uma informação mais concreta. Eu estava testando os endpoints de CEP e CNPJ, para o CEP não precisei mudar nada para realizar o fetch, porém para o CNPJ eu precisei configurar o User-Agent no Header, só assim consegui deixar de receber o erro 403 - Forbidden.

Estou usando Next.JS para testar e meu código ficou mais ou menos assim:

```ts
const response = await fetch(`https://brasilapi.com.br/api/cnpj/v1/${cnpj}`, {
    headers: {
        'User-Agent': 'Biruleibe/1.0',
    }
});
```

Seria interessante se incrementassem a documentação, informando quais headers são necessários para cada endpoint.

Boa sorte!
